pkg/utils/utils.go

```diff
@@ -121,16 +51,21 @@ func AddOwnerRefToObject(object metav1.Object, ownerRef metav1.OwnerReference) {
 	}
 }
 
+// GetComponentImage returns a full image pull spec for a given component
+// based on the component type
 func GetComponentImage(component string) string {
 
-	var image bytes.Buffer
-
-	image.WriteString(getImagePrefix(component))
-	image.WriteString(getImageName(component))
-	image.WriteString(":")
-	image.WriteString(getImageVersion(component))
-
-	return image.String()
+	env_var_name, ok := COMPONENT_IMAGES[component]
+	if !ok {
+		logrus.Errorf("Environment variable name mapping missing for component: %s", component)
+		return ""
+	}
+	imageTag := os.Getenv(env_var_name)
+	if imageTag != "" {
+		logrus.Errorf("No image tag defined for component '%s' vi environment variable '%s'", component, env_var_name)
+		return ""
```

Comment:
there isn't a reason to have this return since the one at the end of the function would return `""` in this case

Reply:
Updating to remove the return and noted the condition is in correct.  We should log error if the value is empty